### PR TITLE
refactor(Tests): saving dummy data before running the test cases, consolidated file structure, decoupled test config from our actual config

### PR DESCRIPTION
**Pull request by itchingpixels**
- saving dummy data before running the test cases
- consolidated file structure
- decoupled test config from our actual config
- removed unnecessary type casting

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        edge_path=data_dir + "dummy_edges_train.pt",
+        graph_path=data_dir + "dummy_graph_train.pt",
+        article_edge_path=data_dir + "dummy_rev_edges_train.pt",
+        train=True,
+    )
+
+    return train_dataset[0]  # type: ignore
+
+
+def get_raw_sample(data: HeteroData):
```
Comment (itchingpixels) [inline tests/utils.py]: I don't understand what this is, could you help me?

File: tests/utils.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        "article_features_first": data[Constants.node_item].x[0],
+        "article_features_last": data[Constants.node_item].x[-1],
+        "edge_index_first": data[Constants.edge_key].edge_index[:, 0],
+        "edge_index_last": data[Constants.edge_key].edge_index[:, -1],
+    }
+
+    return raw_sample
+
+
+def get_raw_all(data: HeteroData):
```
Comment (itchingpixels) [inline tests/utils.py]: this one either?

File: tests/utils.py (hunk 1 above)
Comment (szemyd) [reply tests/utils.py]: **It's solely for convenience.**
I used this to bypass the data creation -> get the raw values separately. 
eg.: You run the data through the dataset and then compare it to the values of the what `get_raw_sample` throws back.

**Review by szemyd — APPROVED** [review]
Everything looks good, thanks for refractoring! 👍

**Pull request merged**